src/checkpoint.rs

```diff
@@ -0,0 +1,228 @@
+use crate::config::CheckpointOptions;
+use crate::db::Db;
+use crate::db_state::Checkpoint;
+use crate::error::SlateDBError;
+use crate::manifest_store::{apply_db_state_update, ManifestStore, StoredManifest};
+use object_store::path::Path;
+use object_store::ObjectStore;
+use std::sync::Arc;
+use std::time::SystemTime;
+
+impl Db {
+    /// Creates a checkpoint of the db stored in the object store at the specified path using the
+    /// provided options. Note that the scope option does not impact the behaviour of this method.
+    /// The checkpoint will reference the current active manifest of the db.
+    pub async fn create_checkpoint(
+        path: Path,
+        object_store: Arc<dyn ObjectStore>,
+        options: &CheckpointOptions,
+    ) -> Result<(uuid::Uuid, u64), SlateDBError> {
```

Comment:
What do you think about returning a struct instead of a tuple? It'd be easier for people to figure out what each value means.

Reply:
Yeah that's fair. I'll make that change.